Github records tests as passing even on failure.

We have noticed that centos-ci test run against PRs for sit-test-environment  are being recorded  as passed even as the console logs show failures. This breaks the CI system and needs to be fixed.

eg: 
PR - https://github.com/samba-in-kubernetes/sit-test-cases/pull/36
![image](https://github.com/samba-in-kubernetes/sit-environment/assets/1053298/c823e5ae-70a0-4c6f-bedd-1b5c8732d871)

This shows gluster tests as passing. But the console logs for gluster tests show the following.
```
=========================== short test summary info ============================
FAILED testcases/smbtorture/test_smbtorture.py::test_smbtorture[replicate-glusterfs-default-smb2.create]
FAILED testcases/smbtorture/test_smbtorture.py::test_smbtorture[disperse-glusterfs-default-smb2.create]
FAILED testcases/mount/test_mount.py::test_mount[192.168.123.10-disperse-glusterfs-default]
FAILED testcases/mount/test_mount.py::test_mount[192.168.123.11-disperse-glusterfs-default]
================== 4 failed, 52 passed in 8039.14s (2:13:59) ===================
ERROR: InvocationError for command /root/sit-test-cases/.tox/pytest/bin/pytest -v testcases/mount testcases/consistency testcases/smbtorture (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   pytest: commands failed

```


@anoopcs9 mentioned that this issue is caused by changes introduced in PR #42 and work is being done to change the way these errors are reported.

--- a/playbooks/ansible/roles/tests/tasks/main.yml
+++ b/playbooks/ansible/roles/tests/tasks/main.yml
@@ -1,4 +1,8 @@
 ---
+- name: Initialize test result
+  set_fact:
+    tests_failed: false
+
 - name: Run tests
   include_tasks: run-test.yml
   vars:
@@ -6,3 +10,8 @@
   loop: "{{ config.tests }}"
   loop_control:
     loop_var: test_name
+
+- name: Check that tests succeeded
+  when: tests_failed
+  fail:
+    msg: "Some tests have failed"

--- a/playbooks/ansible/roles/tests/tasks/run-test.yml
+++ b/playbooks/ansible/roles/tests/tasks/run-test.yml
@@ -31,6 +31,10 @@
         name: "{{ role }}"
         tasks_from: recover/main.yml
 
+    - name: Mark the test as failed
+      set_fact:
+        tests_failed: true
+
   always:
     - name: Log test finalization time
       include_tasks: log.yml

